File: java/Exercise019.java
public class Exercise019 {

    private static String[] weekdays = { "monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday" };
    private static String[] months =
            {"jan", "feb", "march", "april", "may", "jun", "jul", "aug", "sept", "okt", "nov", "dev"};
    private static int[] monthLimits = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };

    public static void main(String[] args)
    {
        int count = 0;
        int dayMonth = 1, dayWeek = 0, month = 0, year = 1900;

        for (int i = 1; i <= 100_000; i++)
        {
            // check if day meets counting criteria
            if (dayWeek == 6 && dayMonth == 1 && year >= 1901 && year <= 2000) count++;

            // continue to next day
            dayWeek = (dayWeek + 1) % 7;
            int limit = monthLimits[month];
            dayMonth++;
            if (dayMonth > limit + leapYearDay(month, year))
            {
                dayMonth = 1;
                month++;
                if (month == 12)
                {
                    month = 0;
                    year++;
                }
            }
            if (year == 2001) break;
        }

        System.out.println(count);
    }

    public static int leapYearDay(int month, int year)
    {
        if (month == 1 && isLeapYear(year)) return 1;
        return 0;
    }

    public static boolean isLeapYear(int year)
    {
        if (year % 400 == 0) return true;
        if (year % 100 == 0) return false;
        if (year % 4 == 0) return true;
        return false;
    }

}

File: python/exercise019.py
def is_leap_year(year):
    if year % 400 == 0:
        return True
    if year % 100 == 0:
        return False
    if year % 4 == 0:
        return True
    return False


def lear_year_day(month, year):
    if month == 1 and is_leap_year(year):
        return 1
    return 0


count = 0
dayMonth = 1
dayWeek = 0
month = 0
year = 1900

monthsLimit = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

for i in range(1,1_000_000):
    if dayWeek == 6 and dayMonth == 1 and 1901 <= year <= 2000:
        count = count + 1

    dayWeek = (dayWeek + 1) % 7
    limit = monthsLimit[month]
    dayMonth = dayMonth + 1
    if dayMonth > limit + lear_year_day(month, year):
        dayMonth = 1
        month = month + 1
        if month == 12:
            month = 0
            year = year + 1
    if year == 2001:
        break

print(count)